packages/headless/src/features/recent-queries/recent-queries-slice.ts

```diff
@@ -45,7 +50,18 @@ export function handleExecuteSearchFulfilled(
   query: string,
   state: WritableDraft<RecentQueriesState>
 ) {
-  state.queries = state.queries.filter((q) => q !== query);
-  const remaining = state.queries.slice(0, state.maxLength - 1);
-  state.queries = [query, ...remaining];
+  const cleanNewQuery = query.trim().toLowerCase();
+  if (cleanNewQuery === '') {
+    return;
+  }
+
+  const previousQueries = Array.from(
```

Comment:
Do we need to preserve the order of the previous queries? If so, the set won't preserve it.

Reply:
Are you sure? I believe JavaScript Set does preserve insertion order

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#description

Reply:
Well, TIL!
<img width="289" alt="image" src="https://github.com/user-attachments/assets/e4c3ecf2-4271-468d-9a90-a7fd1239acaa" />